sdk/servicebus/service-bus/test/internal/batchingReceiver.spec.ts

```diff
@@ -648,21 +648,40 @@ describe("BatchingReceiver unit tests", () => {
 
       assert.notExists(receiver["_closeHandler"]);
 
-      const receiveMessagesPromise = receiver.receiveMessages({
-        maxMessageCount: 1,
-        maxTimeAfterFirstMessageInMs: 1,
-        maxWaitTimeInMs: 1
-      });
+      let resolveWasCalled = false;
+      let rejectWasCalled = false;
+
+      receiver["_receiveMessagesImpl"](
+        (await receiver["_getCurrentReceiver"]())!,
+        {
+          maxMessageCount: 1,
+          maxTimeAfterFirstMessageInMs: 1,
+          maxWaitTimeInMs: 1
+        },
+        () => {
+          resolveWasCalled = true;
+        },
+        () => {
+          rejectWasCalled = true;
+        }
+      );
 
-      await receiveIsReady;
       assert.exists(receiver["_closeHandler"]);
+      assert.isFalse(resolveWasCalled);
+      assert.isFalse(rejectWasCalled);
+
+      receiver.close();
 
-      await receiver.close();
+      // these are still false because we used setTimeout() (and we're using sinon)
+      // so the clock is "frozen"
+      assert.isFalse(resolveWasCalled);
+      assert.isFalse(rejectWasCalled);
 
-      const results = await receiveMessagesPromise;
+      // now unfreeze it (without ticking time forward, just running whatever is eligible _now_)
```

Comment:
Damn!

Reply:
Every time I look into sinon I am just more and more impressed :)